types/react-native-sensor-manager/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for react-native-sensor-manager 0.1
+// Project: https://github.com/kprimice/react-native-sensor-manager
+// Definitions by: Sahin Vardar <https://github.com/SahinVardar>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="react-native" />
+
+export * from 'react-native'
+
+interface SensorManager {
+    startAccelerometer(delay: number): void;
+    stopAccelerometer(): void;
+    startGyroscope(delay: number): void;
+    stopGyroscope(): void;
+    startMagnetometer(delay: number): void;
+    stopMagnetometer(): void;
+    startStepCounter(delay: number): void;
+    stopStepCounter(): void;
+    startThermometer(delay: number): void;
+    stopThermometer(): void;
+    startMotionValue(delay: number): void;
+    stopMotionValue(): void;
+    startOrientation(delay: number): void;
+    stopOrientation(): void;
+    startProximity(delay: number): void;
+    stopProximity(): void;
+    startLightSensor(delay: number): void;
+    stopLightSensor(): void;
+}
+
+declare module 'react-native' {
+    export interface DeviceEventEmitterStatic {
```

Comment:
>I'm not sure if two react-native module declared are conflict.

This would only happen when we have different declarations for the same property inside NativeModules. But then we would also have a problem with the implementation. In this case we would have two react-native native modules which are creating the property inside NativeModules. 

Also before my change, the type of NativeModules is any, so before my change it was not possible to declare properties inside NativeModules.

> A far better approach would be to define a class named SensorManager, however it use NativeModule directly.

According to the official docs, this is a valid approach in react-native.
See [https://facebook.github.io/react-native/docs/native-modules-ios.html](url)